### ISSUE TITLE
Wrong Condition with entity query 

### ISSUE THREAD

**Issue by ahmedwahdan** [thingsboard/thingsboard#3158]
**Describe the bug**
I'm trying to have the following business case:
- Customer has only one user [OK]
- Customer can have many devices [OK]
- Customer can manage other customer devices [Problem]
The problem appears in the dashboard when trying to show the devices "Managed" by current customer, but nothing shows.
I use the following alias 

![image](https://user-images.githubusercontent.com/5406736/88439498-aff12500-ce0b-11ea-8da6-0624371f632d.png)

logged as tenant admin, I can see the devices 

![image](https://user-images.githubusercontent.com/5406736/88439560-d44d0180-ce0b-11ea-8a33-dad7f7965080.png)

I think the problem is with the last condition in the following method:

```
 private static final PermissionChecker customerEntityPermissionChecker =
    new PermissionChecker.GenericPermissionChecker(Operation.READ, Operation.READ_CREDENTIALS,
Operation.READ_ATTRIBUTES, Operation.READ_TELEMETRY, Operation.RPC_CALL, Operation.CLAIM_DEVICES) {

@Override
 public boolean hasPermission(SecurityUser user, Operation operation, EntityId entityId, HasTenantId entity) {

 if (!super.hasPermission(user, operation, entityId, entity)) {
         return false;
     }
     if (!user.getTenantId().equals(entity.getTenantId())) {
          return false;
     }
      if (!(entity instanceof HasCustomerId)) {
            return false;
      }
      f (!operation.equals(Operation.CLAIM_DEVICES) && 
           !user.getCustomerId().equals(((HasCustomerId) entity).getCustomerId())) {
              return false;
      }
               return true;
         }
 };
```
The condition should be :
```
  if (operation.equals(Operation.CLAIM_DEVICES) && 
      !user.getCustomerId().equals(((HasCustomerId) entity).getCustomerId()))
``` 
**Your Server Environment**
Windows 10
Thingsboard v3.0.1

Thanks!

**Comment by ashvarma-git** [comment]
I believe this is same issue #3115 which I posted as question, and it looked like a bug to me but since I am new, I thought I will check first as a question.

**Comment by ashvayka** [comment]
I have fixed the issue here: 2dd2cae. The code line you are referring to is correct. It means: 
"If the operation i am trying to preform is not CLAIM_DEVICES then I should check that customer id from entity matches my customer id". Claim devices is a special operation with additional security checks implemented in the controller. The idea is that customer user can claim devices that belong to tenant and have empty customer id.

**Comment by ashvarma-git** [comment]
Thanks for fixing this issue which I hope also fixes issue #3115 . Can this commit be applied as a patch over existing 3.0.1 CE installation?
